File: generate.js
var fs = require('fs');
var jsonfile = require('jsonfile');
var readYaml = require('read-yaml');
var yaml = require('write-yaml');
var Hexo = require('hexo');


// Файл со списком поддоменов и данными для них
var file = 'subdomains.json'

// Чтение файла и создание массива со всеми данными
var dataSubdomains = jsonfile.readFileSync(file);

// Создание массива только с поддоменами... [moskow, smolensk, и т.д.]
var subdomains = Object.keys(dataSubdomains);
var countSubsomain = subdomains.length;
console.log(subdomains);

var hexo = new Object;
function generateSubdomain(alias) {
    hexo[alias] = new Hexo(process.cwd(), {
        config: "configs/" + alias + ".yml"
    });
    hexo[alias].init().then(function () {
        return hexo[alias].call('generate', {
            watch: false
        });
    }).then(function () {
        return hexo[alias].exit();
    }).then(function () {
        return "все ок"
    }).catch(function (err) {
        console.log(err);
        hexo[alias].exit(err);
    })
}


for (var i = 0; i < subdomains.length; i++) {
    generateSubdomain(subdomains[i]);
}

File: generate.py
# -*- coding: utf-8 -*-

import subprocess
import os
import json

file = 'subdomains.json'

subdomains_json = json.loads(open(file, encoding="utf-8").read())

for subdomain in subdomains_json:
    try:
        os.remove('db.json')
        print('файл db.json удален')
    except(OSError):
        print('файл db.json отсутствует')
    print("Создается контент поддомена: " + subdomain)
    proc = subprocess.Popen("hexo generate --config configs/" + subdomain + ".yml", shell=True, stdout=subprocess.PIPE)
    out = proc.stdout.readlines()